src/main/java/io/jenkins/plugins/coverage/adapter/XMLCoverageReportAdapter.java

```diff
@@ -0,0 +1,40 @@
+package io.jenkins.plugins.coverage.adapter;
+
+import io.jenkins.plugins.coverage.adapter.util.XMLUtils;
+import io.jenkins.plugins.coverage.exception.ConversionException;
+import org.w3c.dom.Document;
+
+import java.io.File;
+import java.net.URISyntaxException;
+
+public abstract class XMLCoverageReportAdapter extends CoverageReportAdapter {
+
+    public XMLCoverageReportAdapter(String path) {
+        super(path);
+    }
+
+    public abstract String getXSL();
```

Comment:
Would be great to document interfaces. E.g. can XSD return `null`? If yes, add `@CheckForNull` annotation and clarify the behavior in Javadoc